File: training/php/src/Adapter/Persistence/CreatureEntity.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Adapter\Persistence;

final class CreatureEntity
{
    public string $id;

    public function __construct(?string $id = null, public string $type = 'monstrosity')
    {
        $this->id = $id ?? 'ididid';
    }

    public function getId(): string
    {
        return $this->id;
    }
}

File: sample-projects/java/src/main/java/de/sots/cellarsandcentaurs/adapter/persistence/CreatureEntity.java
package de.sots.cellarsandcentaurs.adapter.persistence;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import java.util.UUID;

@Entity
@Persisted(table = "creatures")
public class CreatureEntity {
    @Id
    private UUID id;

    private String type;

    public CreatureEntity() {
    }

    public CreatureEntity(UUID id, String type) {
        this.id = id;
        this.type = type;
    }

    public UUID getId() {
        return id;
    }

    public String getType() {
        return type;
    }
}
